tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -217,8 +219,17 @@ public:
 
    RIterator begin();
    RIterator end();
-};
 
+   /// Used for the visitor design pattern, see for example RNTupleReader::Print()
+   virtual void TraverseVisitor(RNTupleVisitor &visitor, int level = 0) const;
+   virtual void AcceptVisitor(RNTupleVisitor &visitor, int level) const;
+   int GetIndex() const {return fNTupleIndex;}
+   bool IsLastInParentSubField() const {
```

Comment:
I agree that a nested struct is a good idea, especially to avoid confusion between fLevel and fOrder.

However I would implement fNFieldsInLevel as GetNumSiblingFields() {return fParent-> fSubfield.size()}. I have a feeling that it's not worth adjusting all fNFieldsInLevel everytime a new sibling-field is attached, as it won't be used very often.

And you once mentioned that for AcceptVisitor it might be good to know the relative level instead of the absolute level. Should I still proceed removing the level parameter? I personally don't see a problem in removing it.

Reply:
Ah, both good points! So yes, please go for `GetNumSiblingFields()` and for leaving the `level` parameter.  There should be a code comment with the explanation why the `level` parameter is necessary.